crates/fuel-core/src/service/sub_services.rs

```diff
@@ -335,6 +345,7 @@ pub fn init_sub_services(
     #[allow(unused_mut)]
     // `FuelService` starts and shutdowns all sub-services in the `services` order
     let mut services: SubServices = vec![
+        Box::new(da_block_costs_service),
```

Comment:
I discussed this with @MitchTurner but it was his opinion that it should run as a sub-service (it was my opinion that the gas price service takes this service as a dependency and should be defined internally)

Reply:
If we define the new service inside of the same crate and it is only used by this crate, I don't see the point in exposing this service outside. We could keep it as an internal detail of the implementation.

@MitchTurner  what are your thoughts?

Reply:
My opposition was with taking a dependency on `RunnableService` inside the `GasPriceService` domain. 

I'm fine with having the adapter manage its own service, OR having a `RunnableService` spun up in integration. But `RunnableService` is an integration level abstraction and I'd prefer it not get used everywhere/anywhere we want to run something in a separate thread.

I honestly don't have problem with it getting spun up on startup just like the `GasPriceService` itself--we already know concrete types when we run `init_sub_services`. I initially suggested we keep it internal to the adapter and ditch the `RunnableService` impl.

Reply:
I like the idea of using `RunnableService` to fetch data from an external source. It looks like what service should be responsible for, since it has some internal logic of how often to fetch data and how to provide this data to its consumers. And this service should support graceful shutdown as well.

I'm okay with having this service as an independent and non-internal sub-service, but then it should live outside of the gas provider crate(currently, it lives there).

I also like the idea of having this service as an internal sub-service of the gas price provider since its usage only belongs to the gas provider. I don't see any problem with having sub-services in the service. The main idea is that each service depends on `fuel-core-services` and is already aware of `RunnableService`. 

At the end of the day, `fuel-core` is also a service with many sub-services =)

Reply:
If you guys both want to use RunnableService inside the other service, then I'll defer to your judgment.

I think the ServiceRunner library is a bit of a Swiss Army knife and that's a design smell for me. But on the other hand, I like having some reusable library for managing services ran in separate threads. I just wish it wasn't trying to solve so many problems at once and the more places we use it the worse that's going to get.

Reply:
integrated in https://github.com/FuelLabs/fuel-core/pull/2163/commits/2f803f9296a96a303f6e5808e7ccc170d55c277b :)